src/components/DataDownloadModal/index.jsx

```diff
@@ -0,0 +1,228 @@
+import Table from 'react-bootstrap/Table';
+
+import Expire from '../Expire';
+import sampledataRegistry from '../../sampledata_registry.json';
+import { getLogger } from '../../logger';
+
+const logger = getLogger(__filename.split('/').slice(-1)[0]);
+
+/** Render a dialog with a form for configuring global invest settings */
+export class DataDownloadModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      allDataCheck: true,
+      allLinksArray: [],
+      selectedLinksArray: [],
+      dataListCheckBoxes: {},
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleCheckAll = this.handleCheckAll.bind(this);
+    this.handleCheckList = this.handleCheckList.bind(this);
+  }
+
+  componentDidMount() {
+    const linksArray = [];
+    const dataListCheckBoxes = {};
+    Object.entries(sampledataRegistry.Models)
+      .forEach(([modelName, data]) => {
+        linksArray.push(data.url);
+        dataListCheckBoxes[modelName] = true;
+      });
+    this.setState({
+      allLinksArray: linksArray,
+      selectedLinksArray: linksArray,
+      dataListCheckBoxes: dataListCheckBoxes,
+    });
+  }
+
+  async handleSubmit(event) {
+    event.preventDefault();
+    const allDataURL = sampledataRegistry.allData.url;
+    // even though the idea is to save files, here we just want to chooose
+    // a directory, so must use OpenDialog.
+    const data = await ipcRenderer.invoke(
+      'show-open-dialog',
+      { properties: ['openDirectory'] }
+    );
+    if (data.filePaths.length) {
+      if (this.state.allDataCheck) {
+        ipcRenderer.send('download-url', [allDataURL], data.filePaths[0]);
+      } else {
+        ipcRenderer.send('download-url', this.state.selectedLinksArray, data.filePaths[0]);
+      }
+      this.props.storeDownloadDir(data.filePaths[0]);
+    }
+    this.props.closeModal();
+  }
+
+  handleCheckAll(event) {
+    const {
+      dataListCheckBoxes,
+      allLinksArray,
+    } = this.state;
+    const newCheckList = Object.fromEntries(
+      Object.entries(dataListCheckBoxes).map(
+        ([k, v]) => [k, event.target.checked]
+      )
+    );
+    let selectedLinks;
+    if (event.target.checked) {
+      selectedLinks = allLinksArray;
+    } else {
+      selectedLinks = [];
+    }
+    this.setState({
+      allDataCheck: event.target.checked,
+      dataListCheckBoxes: newCheckList,
+      selectedLinksArray: selectedLinks,
+    });
+  }
+
+  handleCheckList(event, modelName) {
+    let { selectedLinksArray, dataListCheckBoxes } = this.state;
+    const { url } = sampledataRegistry.Models[modelName];
+    if (event.target.checked) {
+      selectedLinksArray.push(url);
+      dataListCheckBoxes[modelName] = true;
+    } else {
+      selectedLinksArray = selectedLinksArray.filter((val) => val !== url);
+      dataListCheckBoxes[modelName] = false;
+    }
+    this.setState({
+      allDataCheck: false,
```

Comment:
Is this necessarily true? If I have all data selected and then toggle on and off a model, would it be convenient if `allDataCheck` was still `true`?

Reply:
Yeah good question. That would just require checking if all the `dataListCheckBoxes` are `true` on each click. I'm not sure it matters to the user, but there is actually different behavior. If "Select All" is checked, we download one big zipfile that includes everything, as opposed to downloading all the individual folders. I think I did that just because that one big zipfile exists as part of the invest build & deploy process. But that doesn't mean we have to use it.